src/main/java/spoon/support/reflect/declaration/CtElementImpl.java

```diff
@@ -469,6 +469,21 @@ public void replace(CtElement element) {
 		ReplacementVisitor.replace(this, elements);
 	}
 
+	@Override
+	public <E extends CtElement> E setAllMetadata(Map<String, Object> metadata) {
+		if (metadata != null) {
```

Comment:
`if (metadata != null && metadata.size() > 0) ...`
WDYT?

Reply:
or this other way:
```
if (metadata == null) {
  // empty map
  return this;
}
this.metadata.clear();
this.metadata.putAll(metadata);
```